.changeset/unlucky-cameras-joke.md

```diff
@@ -0,0 +1,5 @@
+---
+'@lion/localize': minor
+---
+
+BREAKING: Fires localeChanged event (and as a result, invokes onLocaleChanged / onLocaleUpdated) after localize loading has completed. This means if the user switches the locale to a locale which has not loaded yet, it will load it first before sending the event. This will allow users to immediately call localize.msg and get the right output, without having to await localize.loadingComplete themselves. This is slightly breaking with regards to timing and might break tests in extensions. In that case, you probably need a `await localize.loadingComplete` statement in front of the failing assertion.
```

Comment:
I have a hard time estimating what the breaking impact would be for people consuming LocalizeMixin.

Is there a difference between awaiting `el.localizeNamespacesLoaded` and `localize.loadingComplete`? They all return the same promise in the end, right?
(so for people that used `el.localizeNamespacesLoaded` before, nothing should change). 

Only people that would rely on synchronous (usually mocked) namespaceCache, would need to await an extra promise, now?

Reply:
Yep pretty much. And yeah I do believe `await localize.loadingComplete` or `await el.localizeNamespacesLoaded` do pretty much the same thing, one for the manager and one on the element and whatever namespaces are local to it.

Reply:
Cool. I think the scope is relatively limited then and we can consider it a breaking bugfix 👍